### PR TITLE
Misc Maintenance

### PR DESCRIPTION
**Pull request by domudall**
* Switching to glide for vendoring
* Adding CircleCI configuration to codebase
* Some linting
* Resolving additional race conditions

### CODE REVIEW

File: engine/memory/engine.go [hunk 1]
```diff
@@ -145,10 +145,12 @@ func (e *Engine) Unlock(key string) error {
 func (e *Engine) cleanupExpiredKeys() {
 	go func() {
 		for range time.Tick(e.expirePoll) {
+			storeLock.RLock()
```
Comment (adam-cv) [inline engine/memory/engine.go]: Doesn't `e.IsExpired` also call storeLock.RLock() ?

`https://github.com/fresh8/go-cache/blob/6869bd9196fd080f899679659f38fa035de49340/engine/memory/engine.go#L72-L86`

Comment (domudall) [inline engine/memory/engine.go]: It does yes, however the map is being accessed before the `e.IsExpired` call, and so needs to be read locked before accessing:

https://github.com/fresh8/go-cache/pull/19/files/93588fde355ccf34b67ebdc674c4f56145c63036#diff-573fb8c090b8e2ea3225163826316c93L148

Comment (adam-cv) [inline engine/memory/engine.go]: ok

Comment (domudall) [inline engine/memory/engine.go]: For a demo with/without:

https://circleci.com/gh/fresh8/go-cache/72 - Failure (without read lock)
https://circleci.com/gh/fresh8/go-cache/73 - Success (with read lock)

**Pull request merged**